modules/phase_field/src/userobjects/RandomEulerAngleProvider.C

```diff
@@ -0,0 +1,51 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "RandomEulerAngleProvider.h"
+#include "MooseRandom.h"
+
+template<>
+InputParameters validParams<RandomEulerAngleProvider>()
+{
+  InputParameters params = validParams<EulerAngleProvider>();
+  params.addClassDescription("Assign random euler angles for each grain.");
+  params.addRequiredParam<UserObjectName>("grain_tracker_object", "The FeatureFloodCount UserObject to get values from.");
+  params.addParam<unsigned int>("seed", 0, "Seed value for the random number generator");
+  return params;
+}
+
+RandomEulerAngleProvider::RandomEulerAngleProvider(const InputParameters & params) :
+    EulerAngleProvider(params),
+    _grain_tracker(getUserObject<GrainTrackerInterface>("grain_tracker_object")),
+    _angles(0)
+{
+  MooseRandom::seed(getParam<unsigned int>("seed"));
+}
+
+void
+RandomEulerAngleProvider::initialize()
+{
+  EulerAngles angle;
+  auto grain_num = _grain_tracker.getTotalFeatureCount();
+  for (auto i = _angles.size(); i < grain_num; ++i)
```

Comment:
@SudiptaBiswas - This is perfect! We only need `beginIndex()` when we start a normal loop at zero because the type cannot be inferred from a constant (well it can, but it'll be the wrong type). You've implemented this correctly.